src/include/migraphx/shape.hpp

```diff
@@ -156,8 +156,33 @@ struct shape
 
     shape(const std::vector<shape>& subs);
 
+    /**
+     * Creates an output shape with dimensions equal to the input lengths and strides determined
+     * by the permutation argument.
+     * The "original" dimensions of the shape can be considered to be the forward permutation of the
+     * given output dimensions.
+     * The output shape's strides are the inverse permutation of the standard strides generated by
+     * the "original" dimensions.
+     *
+     * 2D example:
+     *   parameters:
+     *     l = [2, 3], perm = [1, 0]
+     *   therefore:
+     *     inverse_perm = [1, 0]
+     *     original_shape = {lens = [3, 2], strides = [2, 1]}
+     *     output_shape = {lens = [2, 3], strides = [1, 2]
```

Comment:
The lens should be 3, 2, because `from_permutation` doesn't change the order of the lens.

Reply:
This is correct, same as from_2d_permutation testcase.

Reply:
In the test `out_shape.lens() == out_lens` so the order of the dimensions do not change.

Reply:
Yeah, the input `l` is [2, 3] the output_shape has lens = [2, 3]. What I mean by "original" shape is the un-transposed shape.